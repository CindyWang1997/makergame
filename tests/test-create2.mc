extern void end_game();

helper {
  // should run before main's second step
  step { printstr("helper"); }
}

aide {
  // should run before main's second step
  step { printstr("aide"); }
}

main {
  bool created;
  create { this.created = false; create aide; }
  step {
    printstr("main");
    if (!this.created) {
      create helper;
      create helper;
      this.created = true;
    }
    else {
      create aide; // causes an 'aide' call right before the end
      end_game();
    }
  }
}

