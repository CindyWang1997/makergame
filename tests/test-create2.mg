
object helper {
  // should run before main's second step
  event step { std::print::s("helper"); }
}

object aide {
  // should run before main's second step
  event step { std::print::s("aide"); }
}

object main {
  bool created;
  event create { this.created = false; create aide; }
  event step {
    std::print::s("main");
    if (!this.created) {
      create helper;
      create helper;
      this.created = true;
    }
    else {
      create aide; // causes an 'aide' call right before the end
      std::game::end();
    }
  }
}

