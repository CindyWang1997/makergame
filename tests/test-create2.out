main
aide
helper
helper
main
aide
helper
helper
aide
